ARTstor.js

```diff
@@ -34,98 +37,1043 @@
+    } else {
+        return false;
+    }
+
 }
 
-function associateData(newItem, value, field) {
-	if (!value) return;
-	if (typeof value == 'string') value = ZU.trimInternal(value);
-	if (!value) return;
-	
-	newItem[field] = value;
+function decodeSearchData(str) {
+    var searchParam = str;
+    searchParam = searchParam.replace(/(&gt;)/, ":");
+    var param = searchParam.split('&');
+    var searchData = new Object();
+    var item;
+    var sparam;
+    var id;
+    var value;
+    for (var i = 0; i < param.length; i = i + 1) {
+        item = param[i];
+        sparam = item.split('=');
+        id = sparam[0];
+        value = sparam[1];
+
+        if (id === 'type') {
+            searchData[id] = parseInt(value);
+        } else
+            searchData[id] = value;
+    }
+    return searchData;
 }
 
 function doWeb(doc, url) {
-	var newItem = new Zotero.Item("artwork"),
-		data = getData(doc),
-		content;
-	
-	if (data.Creator) {
-		content = data.Creator.replace(/\s*\(.*/, '');
-		associateData(newItem,
-			ZU.cleanAuthor(content, "artist", artist.indexOf(',') != -1),
-			'creators'
-		);
-	}
-	
-	if (data.Date) {
-		content = data.Date.replace(/^Photographed (?:in )?/, '');
-		associateData(newItem, content, "date");
-	}
-	
-	if (data.Rights) {
-		var m = data.Rights.match(/^.*(©.*)$/m);
-		if (m) associateData(newItem, m[1], "rights");
-	}
-	
-	if (data.Subject) {
-		content = data.Subject.split(/\s*;\s*/);
-		associateData(newItem, content, 'tags');
-	}
-	
-   //these might not be complete - it's pretty straightforward to add more
-	associateData(newItem, data.Title, "title");
-	associateData(newItem, data.Measurements, "artworkSize");
-	associateData(newItem, data.Material, "artworkMedium");
-	associateData(newItem, data.Repository, "archive");
-	associateData(newItem, data['ID Number'], "archiveLocation");
-	associateData(newItem, data.Description, "abstractNote");
-	
-	newItem.complete();
-}
\ No newline at end of file
+    var docType = detectWeb(doc, url);
+    if (docType == "multiple") {
+        scrape(doc, url);
+    } else if (docType == "artwork") {
+        if (url.match(/iv2/)) {
+            scrapeIV(doc, url);
+        } else {
+            scrapeMetaWindow(doc, url);
+        }
+    }
+}
+
+
+var decodeMap = {
+    '20': ' ',
+    '21': '!',
+    '22': '"',
+    '23': '#',
+    '24': '$',
+    '25': '%',
+    '26': '&',
+    '27': "'",
+    '28': '(',
+    '29': ')',
+    '2A': '*',
+    '2B': '+',
+    '2C': ',',
+    '2D': '-',
+    '2E': '.',
+    '2F': '/',
+    '30': '0',
+    '31': '1',
+    '32': '2',
+    '33': '3',
+    '34': '4',
+    '35': '5',
+    '36': '6',
+    '37': '7',
+    '38': '8',
+    '39': '9',
+    '3A': ':',
+    '3B': ';',
+    '3C': '<',
+    '3D': '=',
+    '3E': '>',
+    '3F': '?',
+    '5B': '[',
+    //				'5C' : '\\',
+    '5D': ']',
+    '5E': '^',
+    '5F': '_',
+    '60': '`',
+    '7B': '{',
+    '7C': '|',
+    '7D': '}',
+    '7E': '~'
+};
+
+function decrypt(s) {
```

Comment:
Can you explain what you're trying to decrypt/encrypt here? Looks to me like at least for part of it you're just doing [`decodeURIComponent`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/decodeURIComponent)/[`encodeURIComponent`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/encodeURIComponent)